### PR TITLE
Tools: Fix OSX environment install script for Big Sur

### PR DESCRIPTION
**Pull request by jmachuca77**
Fixed issues with brew tap
Install using python3
Add python3 local to path
give option to add path to bash or zsh shell

### CODE REVIEW

**Comment by jmachuca77** [comment]
Finally had some time last night to finish this. 

- I had to add pyenv to install python 3.9.1 as the Apple installation does not work correctly for us. 
- Installed python 3.9.1 (framework version) using pyenv and set it as the global version
- Had to install wxpython using brew because the one installed by pip complains about not having access to the display.
- Added -H flag to pip install to avoid warnings
- Added support for both shells (bash, zsh)
- Cleaned up the message output.
- Tested on a fresh install of Big Sur on parallels virtual machine. If someone else could test it would be appreciated.

File: Tools/environment_install/install-prereqs-mac.sh [hunk 1]
```diff
@@ -1,47 +1,93 @@
 #!/bin/bash
-echo "Checking homebrew..."
+
+echo "`tput setaf 2`Checking homebrew...`tput sgr0`"
```
Comment (peterbarker) [inline Tools/environment_install/install-prereqs-mac.sh:3]: Can we create a "heading" function rather than putting all of these `tput`'s inline, please?

**Comment by rmackay9** [comment]
As discussed on the dev call, let's try and get another mac users to test this change before we merge.

**Comment by andyp1per** [comment]
I'm trying this on a Big Sur mac that has some environment already. Here are the problems I noticed:

- I needed xcode install of the command line tools - it does this for me (good) but then continues even though later steps require the xcode tools
- Re-runnning I get some errors:
```
Error: Formulae found in multiple taps: 
       * ardupilot/ardupilot/gcc-arm-none-eabi
       * ardupilot/px4/gcc-arm-none-eabi
```
```
Installing Python 3.9.1 and setting it as pyenv global
python-build: use openssl@1.1 from homebrew
python-build: use readline from homebrew
Downloading Python-3.9.1.tar.xz...
-> https://www.python.org/ftp/python/3.9.1/Python-3.9.1.tar.xz
Installing Python-3.9.1...
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 11.2.2 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/vw/0z3td4td53l6j48m5fzq5jxw0000gn/T/python-build.20210312153036.20807
Results logged to /var/folders/vw/0z3td4td53l6j48m5fzq5jxw0000gn/T/python-build.20210312153036.20807.log

Last 10 log lines:
checking MACHDEP... "darwin"
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... configure: error: in `/var/folders/vw/0z3td4td53l6j48m5fzq5jxw0000gn/T/python-build.20210312153036.20807/Python-3.9.1':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
make: *** No targets specified and no makefile found.  Stop.
pyenv: version `3.9.1' not installed
```
I think you may need a better configuration for pyenv, my pyenv experience on macOS is not good.
```
==> Installing wxpython dependency: python@3.9
==> Pouring python@3.9-3.9.2_1.big_sur.bottle.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/pip3
Target /usr/local/bin/pip3
already exists. You may want to remove it:
  rm '/usr/local/bin/pip3'

To force the link and overwrite all conflicting files:
  brew link --overwrite python@3.9

To list all files that would be deleted:
  brew link --overwrite --dry-run python@3.9
```
```
==> Installing wxpython dependency: numpy
==> Pouring numpy-1.20.1.big_sur.bottle.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/f2py
Target /usr/local/bin/f2py
already exists. You may want to remove it:
  rm '/usr/local/bin/f2py'

To force the link and overwrite all conflicting files:
  brew link --overwrite numpy

To list all files that would be deleted:
  brew link --overwrite --dry-run numpy

Possible conflicting files are:
/usr/local/bin/f2py
/usr/local/bin/f2py3
```
```
Warning: wxpython dependency gcc was built with a different C++ standard
library (libstdc++ from clang). This may cause problems at runtime.
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/helpviewer
Target /usr/local/bin/helpviewer
already exists. You may want to remove it:
  rm '/usr/local/bin/helpviewer'

To force the link and overwrite all conflicting files:
  brew link --overwrite wxpython

To list all files that would be deleted:
  brew link --overwrite --dry-run wxpytho
```

**Comment by peterbarker** [comment]
@jmachuca77 any ideas here?  Sounds like it wasn't butter-smooth for @andyp1per

**Comment by jmachuca77** [comment]
Seems like he already had modifications to the OS X base install and had already installed several of the dependencies. I would not want our scripts to overwrite someones existing installs. As he had already installed the PX4 tap in brew he is getting the error that a package exits in both. I could either try and remove the existing PX4 tap, or see if there is a way to force it to install from our tap, I'll look into that. 

@andyp1per what do you mean by this:
"I needed xcode install of the command line tools - it does this for me (good) but then continues even though later steps require the Xcode tools"
if it installs the command line tools, why do you mention it continues? if it installed them what is the issue you are getting?

**Comment by jmachuca77** [comment]
@andyp1per ?

**Comment by andyp1per** [comment]
I think it didn't actually install the tools, just told me I needed to and then proceeded anyway.

**Comment by jmachuca77** [comment]
@andyp1per looks like my test VM already has the tools installed so it just skips over the install... I'll have to make a new vm without the command line tools installed.

**Comment by patrickelectric** [comment]
This is open since february and there are people that needs this fixed, can we just get https://github.com/ArduPilot/ardupilot/pull/17405 merged ? Is just a single line that changes the gcc version. 
Otherwise, is there a timeline to get it merged ?

**Comment by jmachuca77** [comment]
I tested this on my setup again and it worked without any issues again. I think it should be merged.

**Comment by bugobliterator** [comment]
Damn, wasn't aware that this was in the works! I did another version: https://github.com/ArduPilot/ardupilot/pull/17860 if there's anything that can be learnt from there.

**Comment by jmachuca77** [comment]
Closing, superseded by #17860

**Pull request closed without merging**